ACPs/176-dynamic-evm-gas-limit-and-price-discovery-updates/README.md

```diff
@@ -0,0 +1,138 @@
+
+### Configuration Parameters
+As noted above, the gas price determination mechanism depends on the values of $T$, $M$, $K$, $C$, and $R$ to be set as parameters. $T$ is adjusted dynamically from its initial value based on $D$ and $P$, and the values of $R$ and $C$ are derived from $T$. 
+
+Parameters at activation on the C-Chain are:
+
+<div align="center">
+
+| Parameter | Description | P-Chain Configuration|
+| - | - | - |
+| $T$ | initial target gas consumed per second | $1,500,000$ |
+| $M$ | minimum gas price | $1*10^{-18}$ AVAX  |
+| $K$ | gas price update constant | $2,164,043$ |
+| $D$ | target change bound divisor | $1024$ |
+| $P$ | minimum target gas consumption per second | $1,000,000$
+
+</div>
+
+$T$ was chosen as the current target gas consumption rate on the C-Chain. Avalanche L1s may change this value to match their current gas consumption rate if they would like.
+
+$M$ was chosen as the minimum possible denomination of the native EVM asset, such that the gas price will be more likely to consistently be in a range of price discovery. The price discovery mechanism has already been battle tested on the P-Chain (and prior to that on Ethereum for blob gas prices as defined by EIP-4844), giving confidence that it will correct react to any increase in network usage in order to prevent a DOS attack.
+
+$K$ was chosen such that at sustained maximum capacity ($T*2$ gas/second), the fee rate will double every ~30 seconds.
+
+$D$ was chosen to match the [gas limit bound divisor that Ethereum currently uses](https://github.com/ethereum/go-ethereum/blob/52766bedb9316cd6cddacbb282809e3bdfba143e/params/protocol_params.go#L26) to limit the amount that validators can change their execution layer gas limit.
+
+$P$ was chosen as a safe bound on the minimum target gas usage that is lower than the current target gas consumption rate on the C-Chain today. The target gas consumption rate will not move towards $P$ unless the majority of stake weight of the network specifies $P$ as their desired gas consumption rate target.
+
+## Backwards Compatibility
+
+The changes proposed in this ACP require a required network upgrade in order to take effect. Prior to its activation, the current gas limit and price discovery mechanisms will continue to be used. Its activation should have relatively minor compatibility effects on any developer tooling. Notably, transaction formats, and thus wallets, are not impacted. After its activation, given that the value of $C$ is dynamically adjusted, the maximum possible gas consumed by an individual block, and thus maximum possible consumed by an individual transaction, will also dynamically adjust. This can affect that maximum size of smart contract that can be deployed at any given time, but should not have significant practical impact.
```

Comment:
The maximum size of smart contract that can be deployed at any given time should not have significant practical impact? 👀

Reply:
Ah, worth noting that the static contract size limit is still in place. It's just that some contract deployments take significant gas usage, primarily when they include complex constructor/initialization logic. So nothing specific to contracts necessarily, it's just that the maximum gas that a single tx can use will fluctuate, and contract deployments can tend to use significant gas.

Reply:
I'll reword to clarify

Reply:
Ahh okay :)